lib/internal/errors.js

```diff
@@ -574,6 +574,22 @@ function oneOf(expected, thing) {
   }
 }
 
+function addNumericalSeparator(val) {
+  let res = val[val.length - 1];
+  let count = 0;
+  for (let i = val.length - 2; i >= 0; i--) {
+    count++;
+    if (val[i] === '.') {
+      return val;
+    }
+    if (count % 3 === 0 && i !== val.length && val[i] !== '-') {
+      res = `_${res}`;
+    }
+    res = `${val[i]}${res}`;
+  }
+  return res;
+}
```

Comment:
Can I suggest a variant that's a bit briefer and (hopefully) a bit faster?
```js
function addNumericalSeparator(s) {
  if (s.includes('.')) return s;
  const rc = [];
  const start = +(s[0] === '-');
  const len = s.length - start;
  let pos = start + len % 3;
  if (pos > start) rc.push(s.slice(0, pos));
  for (/* empty */; pos < s.length; pos += 3) rc.push(s.slice(pos, pos + 3));
  return rc.join('_');
}
```

Reply:
In a mini benchmark this is around 50% slower than my version.

Reply:
I changed the implementation and it's now twice as fast as my former version.

Reply:
Sorry, missed your reply. The reason I suggested `Array#join()` is that it creates a flat string whereas a template string creates (maybe I should say: can create) a cons string that needs to be flattened later on. IOW, the cost center shifts away from this function but the cost is still there, it's just more diffuse.

It probably doesn't matter that much for short strings however so this is fine.